### PR TITLE
echelon: add isolated pairs besides the main pool

### PR DESCRIPTION
**Pull request by 0xJ3**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### methodology (what is being counted as tvl, how is tvl being calculated):

echelon consists of 2 parts: a main pool (think compound v2) and isolated pairs (think fraxlend).
- for main pool, tvl = sum(all coins supplied)
- for isolated pairs, tvl = sum(all collateral tokens) + sum(all asset tokens supplied). see [fraxlend](https://docs.frax.finance/fraxlend/fraxlend-overview) for reference

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/echelon: 
        
 
  
    
    
    TypeError: Cannot read properties of undefined (reading 'map')
    Truncated error stack:
    
        at getIsolatedPairAddresses (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/echelon/index.js:27:51)
        at async borrowed (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/echelon/index.js:76:37)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:171:9
        at async Promise.all (index 1)
        at async Promise.all (index 2)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:214:3

**Comment by llamatester** [comment]
Error while running adapter at projects/echelon: 
        
 
  
    
    
    Error: Invalid balance: NaN
    Truncated error stack:
        at isValidNumber (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/generalUtil.js:114:19)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:344:16
        at Array.forEach (<anonymous>)
        at fixBalances (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:338:28)
        at computeTVL (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:351:3)
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:54:30)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:171:9
        at async Promise.all (index 1)

**Comment by llamatester** [comment]
The adapter at projects/echelon exports TVL: 
        
 
 
    aptos-borrowed            98.35 M
    borrowed                  98.35 M
    aptos                     40.89 M
    
    total                    40.89 M

**Comment by llamatester** [comment]
The adapter at projects/echelon exports TVL: 
        
 
 
    aptos-borrowed            103.20 M
    borrowed                  103.20 M
    aptos                     41.30 M
    
    total                    41.30 M

File: projects/echelon/index.js [hunk 1]
```diff
@@ -29,6 +52,15 @@ module.exports = {
         sdk.util.sumSingleBalance(balances, coin, cash);
       });
 
+      const isolatedPairAddresses = await getIsolatedPairAddresses();
+      const pairs = await Promise.all(isolatedPairAddresses.map(pairAddress => getPair(pairAddress)));
+      pairs.forEach(({ collateral_token, total_collateral_amount, asset_token, total_supply_amount }) => {
+        // isolated pair collateral
+        sdk.util.sumSingleBalance(balances, collateral_token, total_collateral_amount);
+        // isolated pair supply
+        sdk.util.sumSingleBalance(balances, asset_token, total_supply_amount);
```
Comment (g1nt0ki) [inline projects/echelon/index.js:61]: you need to exclude the borrowed amount here

**Comment by llamatester** [comment]
The adapter at projects/echelon exports TVL: 
        
 
 
    aptos-borrowed            103.84 M
    borrowed                  103.84 M
    aptos                     41.79 M
    
    total                    41.79 M

File: projects/echelon/index.js (hunk 1 above)
Comment (0xJ3) [reply projects/echelon/index.js]: @g1nt0ki addressed

**Pull request merged**